Upgrade to intervantion v3

tauseefsshah opened the pull request:
This PR aims to fix #1187  using laravel's `response()` method and upgrade intervention to `v3`.

tauseefsshah commented:
@driesvints The functionality works but I am not able fix the test that I wrote. The test runs fine on my local (MacOS + Herd) but fails on github actions. I suspect it's because of different environments or something similar but not sure. Should I remove the test for now so you can merge this PR?

driesvints commented:
@tauseefsshah heya sorry for the long wait here.. could you maybe rebase this PR and fix the conflicts before I review? Thanks!

driesvints commented on package.json:
```diff
@@ -20,5 +20,8 @@
         "vite": "^5.0.12"
+    },
+    "dependencies": {
+        "pixelmatch": "^5.0.0"
```
Move this to devdependencies

driesvints reviewed (changes requested):
Looks real good overall!

driesvints commented on app/Jobs/GenerateSocialShareImage.php:
```diff
@@ -23,16 +25,19 @@ final class GenerateSocialShareImage
+                    $font->color(self::TEXT_COLOUR);
+                })
+                ->toPng()
+        )->header('Content-Type', 'image/png');
```
Would be good if we could cache this somehow as well.

tauseefsshah commented:
Hey @driesvints 👋 Made the following updates:

1. Rebased the branch with the latest `main` branch
2. Moved `pixelmatch` to devDeps 
3. Added cache for the response using Laravels `Cache` with a ttl set to 24 hours

Please review now and also what about the test I wrote? It's still failing on github actions. Should I skip/remove the test for now?

driesvints commented:
I disabled that test for now. Couldn't figure it out either but it's running locally and I verified this approach works.

Thanks a bunch for your work here @tauseefsshah! Sorry it took me so long to review this.

driesvints commented:
I now see the text is just a little too high. Could you adjust that maybe @tauseefsshah ?

<img width="580" alt="Scherm­afbeelding 2025-03-07 om 08 33 26" src="https://github.com/user-attachments/assets/a976bbd7-5d15-4e8a-b30a-6c81ae1c784c" />

Should be more like this:

<img width="591" alt="Scherm­afbeelding 2025-03-07 om 08 35 39" src="https://github.com/user-attachments/assets/94695efe-f768-4b5e-aa46-a1c5b723ec9d" />

tauseefsshah commented:
Hey @driesvints 
Yes let me fix this and raise a PR :)

tauseefsshah commented:
Hey @driesvints Have raised a PR #1262 to fix the positioning
